salt/modules/disk.py

```diff
@@ -580,3 +586,130 @@ def smart_attributes(dev, attributes=None, values=None):
         smart_attr[attr] = data
 
     return smart_attr
+
+
+@depends(HAS_IOSTAT)
+def iostat(interval=1, count=5, disks=None):
+    '''
+    Gather and return (averaged) IO stats.
+
+    .. versionadded:: Boron
+
+    CLI Example:
+    .. code-block:: bash
+        salt '*' disk.iostat 1 5 disks=sda
+    '''
+    if salt.utils.is_linux():
+        return _iostat_linux(interval, count, disks)
+    elif salt.utils.is_freebsd():
+        return _iostat_fbsd(interval, count, disks)
+
+
+def _iostats_dict(header, stats):
+    '''
+    Transpose collected data, average it, stomp it in dict using header
+
+    Use Decimals so we can properly calc & round, convert to float 'caus' we can't transmit Decimals over 0mq
+    '''
+    stats = [float((sum(stat) / len(stat)).quantize(decimal.Decimal('.01'))) for stat in zip(*stats)]
+    stats = dict(zip(header, stats))
+    return stats
+
+
+def _iostat_fbsd(interval, count, disks):
+    '''
+    Tested on FreeBSD, quite likely other BSD's only need small changes in cmd syntax
+    '''
+    if disks is None:
+        iostat_cmd = 'iostat -xC -w {0} -c {1} '.format(interval, count)
+    elif isinstance(disks, six.string_types):
+        iostat_cmd = 'iostat -x -w {0} -c {1} {2}'.format(interval, count, disks)
+    else:
+        iostat_cmd = 'iostat -x -w {0} -c {1} {2}'.format(interval, count, ' '.join(disks))
+
+    sys_stats = []
+    dev_stats = collections.defaultdict(list)
+    sys_header = []
+    dev_header = []
+    h_len = 1000  # randomly absurdly high
+
+    ret = iter(__salt__['cmd.run_stdout'](iostat_cmd, output_loglevel='quiet').splitlines())
+    for line in ret:
+        if not line.startswith('device'):
+            continue
+        elif not len(dev_header):
+            dev_header = line.split()[1:]
+        while True:
```

Comment:
Hi @cachedout! the top `for` loop already takes ret, so I'm stuck with using `next()`. Also, `iostats` output isn't particularly helpful:

```
Linux 4.2.6-301.fc23.x86_64 (lianxiang.the-loeki.net) 	08/12/15 	_x86_64_	(4 CPU)

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           5.30   22.68    1.92    1.32    0.00   68.78

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
sda               0.00    13.88    8.51    4.02    63.04   379.42    70.61     0.24   18.92    4.09   50.29   1.31   1.64
sdb              26.45    36.58  219.11   11.86  2039.91   326.14    20.49     0.11    0.46    0.38    2.00   0.13   3.02
bcache0           0.00     0.00  188.72   55.04  1381.92   571.20    16.02     8.45   34.65    0.39  152.13   0.15   3.72

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.01   25.13    1.51    0.25    0.00   72.11

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
sda               0.00     0.00    0.00    1.00     0.00     4.00     8.00     0.00    0.00    0.00    0.00   0.00   0.00
sdb               0.00     0.00   45.00    0.00   172.00     0.00     7.64     0.01    0.13    0.13    0.00   0.13   0.60
bcache0           0.00     0.00   42.00    0.00   168.00     0.00     8.00     0.01    0.12    0.12    0.00   0.12   0.50

```

As you can see there's little in the way of distincting when the Device section ends, other than 'keep looping until you find something non-alphanumeric on char 0 OR the `next()` iterator is empty' (e.g. you parsed the entire output, preventing infinite loops).

Now that I look at it, I could add an additional bomb by changing to `while line is not False`?